lib/src/visitor/async_evaluate.dart

```diff
@@ -1469,7 +1469,14 @@ class _EvaluateVisitor
         await _environment.withContent(contentCallable, () async {
           await _environment.asMixin(() async {
             for (var statement in mixin.declaration.children) {
-              await statement.accept(this);
+              try {
+                await statement.accept(this);
+              } on SassRuntimeException catch (error) {
+                if (error.type != SassRuntimeExceptionType.error) rethrow;
+
+                throw SassRuntimeException(
+                    error.message, node.span, _stackTrace(error.span));
+              }
```

Comment:
Since this `try`/`catch` is repeated twice, consider putting it in a helper method, similar to the existing `_addExceptionSpanAsync` helper.